cmd/util/ledger/util/payload_file.go

```diff
@@ -22,13 +25,108 @@ const (
 	payloadEncodingVersion = 1
 )
 
+const (
+	MagicBytesPayloadHeader uint16 = 0x2138
+
+	PayloadFileVersionV1 uint16 = 0x01
+
+	encMagicBytesIndex       = 0
+	encMagicBytesSize        = 2
+	encVersionIndex          = 2
+	encVersionSize           = 2
+	encFlagIndex             = 4
+	encFlagLowByteIndex      = 5
+	encPartialStateFlagIndex = encFlagLowByteIndex
+	encFlagSize              = 2
+	headerSize               = encMagicBytesSize + encVersionSize + encFlagSize
+	encPayloadCountSize      = 8
+	footerSize               = encPayloadCountSize
+	crc32SumSize             = 4
+)
+
+const (
+	maskPartialState byte = 0b0000_0001
+)
+
+// newPayloadFileHeader() returns payload header, consisting of:
+// - magic bytes (2 bytes)
+// - version (2 bytes)
+// - flags (2 bytes)
```

Comment:
Yes, 1 byte for flag is enough for now.  But I ran into a situation in onflow/atree where all flag bits got used, so I use 2 bytes here to avoid this potential issue since it is only 1 extra byte written only once to a file.